src/main/java/hudson/plugins/sshslaves/verifiers/KnownHostsFileKeyVerificationStrategy.java

```diff
@@ -73,6 +74,16 @@ public boolean verify(SlaveComputer computer, HostKey hostKey, TaskListener list
         }
         
     }
+
+    @Override
+    public String[] getPreferredKeyAlgorithms(SlaveComputer computer) throws IOException {
+        if (!KNOWN_HOSTS_FILE.exists()) {
+            return super.getPreferredKeyAlgorithms(computer);
+        }
+
+        KnownHosts knownHosts = new KnownHosts(KNOWN_HOSTS_FILE);
+        return knownHosts.getPreferredServerHostkeyAlgorithmOrder(((SSHLauncher) computer.getLauncher()).getHost());
```

Comment:
Unchecked cast. Please avoid it, because there is a risk somebody uses API wrong. And FindBugs will start grumbling once enabled